### PR TITLE
Conversion to Custom Data from EIDF

### PR DESCRIPTION
**Pull request by Vapok**
This is the PR of the Custom Data from EIDF.

Test Cases Included:
1) Existing Gear in Inventory shows as magical.
2) Existing Gear on Ground shows as magical with loot beams.
3) spawn SwordIron - with magicmats - and Enchanted Item goes from mundane to magical.
4) Augment a property off of Enchanted Item from old EIDF world.
5) Augment a property off of Enchanted Item from new non-EIDF world.
6) mi Legendary SwordIron
7) misets Ragnar
8) Logged in and out, data saved.

### CODE REVIEW

File: EpicLoot/LootRoller.cs [hunk 1]
```diff
@@ -262,6 +255,7 @@ private static List<GameObject> RollLootTableInternal(LootTable lootTable, int l
                             {
                                 foreach (var itemAmountConfig in disenchantProducts)
                                 {
+                                    EpicLoot.Log($"[RollLootInternal] Step 9 - {ld.Item} - Amount: {itemAmountConfig.Amount}");
```
Comment (RandyKnapp) [inline EpicLoot/LootRoller.cs:258]: Lets remove logs like these if they're not relevant after testing. Is this something that anyone needs to know on a regular basis?

File: EpicLoot/LootRoller.cs [hunk 2]
```diff
@@ -196,15 +194,13 @@ private static List<GameObject> RollLootTableInternal(LootTable lootTable, int l
             {
                 return results;
             }
-
```
Comment (RandyKnapp) [inline EpicLoot/LootRoller.cs:199]: Why did you take out these blank lines? Blank lines are useful to create breaks between groups. The comment I made to you earlier was for extraneous line breaks (like at the beginning or end of a block) and double line breaks.

File: EpicLoot/EpicLoot.cs [hunk 3]
```diff
@@ -232,8 +232,8 @@ private void Awake()
 
             LoadAssets();
 
-            ExtendedItemData.LoadExtendedItemData += MagicItemComponent.OnNewExtendedItemData;
-            ExtendedItemData.NewExtendedItemData += MagicItemComponent.OnNewExtendedItemData;
+            //ExtendedItemData.LoadExtendedItemData += MagicItemComponent.OnNewExtendedItemData;
```
Comment (RandyKnapp) [inline EpicLoot/EpicLoot.cs:235]: I think we can go ahead and remove the commented lines, git will save the history

**Review by RandyKnapp — CHANGES_REQUESTED** [review]

File: EpicLoot/CharacterDrop_Patch.cs [hunk 4]
```diff
@@ -85,13 +86,15 @@ public static void Postfix(CharacterDrop __instance, ref List<KeyValuePair<GameO
...
+
                     var itemDrop = prefab.GetComponent<ItemDrop>();
+
                     if (itemDrop == null)
                     {
                         continue;
                     }
 
-                    if (itemDrop.m_itemData?.m_shared.m_itemType == ItemDrop.ItemData.ItemType.Trophie)
+                    if (itemDrop.m_itemData.m_shared.m_itemType == ItemDrop.ItemData.ItemType.Trophie)
```
[the remark this answers is not in the document]
Comment (Vapok) [reply EpicLoot/CharacterDrop_Patch.cs:97]: Nice catch...  I'm not sure why that got removed.. I'm also going to just check for == null on that data as well.

File: EpicLoot/EpicLoot.cs (hunk 3 above)
Comment (Vapok) [reply EpicLoot/EpicLoot.cs]: Done.

File: EpicLoot/CharacterDrop_Patch.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vapok) [reply EpicLoot/CharacterDrop_Patch.cs:97]: Done.

File: EpicLoot/LootRoller.cs (hunk 2 above)
Comment (Vapok) [reply EpicLoot/LootRoller.cs]: Put spaces back in. :)

File: EpicLoot/LootRoller.cs (hunk 1 above)
Comment (Vapok) [reply EpicLoot/LootRoller.cs]: Removed.

**Review by RandyKnapp — APPROVED** [review]

**Comment by AnnanFay** [comment]
This pull request as far as I have found it probably the best explanation of how to convert ExtendedItemDataFramework to the new method.

Just putting some strings in here so it appears in google search a bit easier - took a while to find it.

"IF ANOTHER MOD REQUIRES THIS MOD, ASK THE AUTHOR TO CONVERT TO USING CUSTOM ITEMDATA."

backlink: https://thunderstore.io/c/valheim/p/RandyKnapp/ExtendedItemDataFramework/

For some reason finding information about this new format is difficult to find. With quotes it only gives the thunderstore page, without quotes it gives only results before 2021 or which are trying to add new items (full items not custom data).

firefox_iNadxAI3JP

**Pull request merged**